pydarn/plotting/rtp.py

```diff
@@ -0,0 +1,942 @@
+            Default: viridis
+            note: to reverse the color just add _r to the string name
+        boundary: dict
+            dict of the parameter names of a summary plot as key names:
+                - 'search.noise': search noise
+                - 'sky.noise': sky noise
+                - 'nave': number of averages
+                - 'tfreq': transmission frequency
+                - 'p_l': Signal to Noise
+                - 'v': velocity
+                - 'w_l': spectral width
+                - 'elv': elevation
+            with a tuple as the value (zmin, zmax) for the plots
+            Default: {'noise.search': (1e0, 1e5),
+                      'noise.sky': (1e0, 1e5),
+                      'tfreq': (8, 22),
+                      'nave': (0, 60),
+                      'p_l': (0, 30),
+                      'v': (-200, 200),
+                      'w_l': (0, 150),
+                      'elv': (0, 50)}
+        plot_elv: boolean
+            boolean determines if elevation should be plotted or not.
+            If there is no data for elevation data field then elevation is not
+            plotted.
+            Default: True
+        title: str
+            title of the plot
+            Default: auto-generated by the files details
+            {radar name} Fitacf {version} {start_date} - {end_date}  Beam {num}
+        Raises
+        ------
+        IndexError
+        RTPUnknownParameterError
+        RTPIncorrectPlotMethodError
+        RTPNoDataFoundError
+
+        See Also
+        --------
+        plot_range_time : plots range-time parameter
+        plot_time_series : plots time-series
+
+        Return
+        ------
+        fig: matplotlib.pyplot.figure object
+        axes: list
+            list of matplotlib.pyplot.axes objects to generate
+            the subplots in the summary plot
+        """
+        message = "WARNING: matplotlib Default dpi may cause distortion"\
+                  " in range gates and time period. The figure size can"\
+                  " be adjusted with the option figsize and dpi can be"\
+                  " adjusted when saving the file."
+        warnings.warn(message)
+
+        # Default boundary ranges for the various parameter
+        boundary_ranges = {'noise.search': (1e0, 1e5),
+                           'noise.sky': (1e0, 1e5),
+                           'tfreq': (8, 22),
+                           'nave': (0, 60),
+                           'p_l': (0, 30),
+                           'v': (-200, 200),
+                           'w_l': (0, 150),
+                           'elv': (0, 50)}
+        boundary_ranges.update(boundary)
+
+        # Default color maps for the summary plot
+        cmap = {'noise.search': 'k',
+                'noise.sky': 'k',
+                'tfreq': 'k',
+                'nave': 'k',
+                'p_l': 'viridis',
+                'v': 'viridis',
+                'w_l': 'viridis',
+                'elv': 'viridis'}
```

Comment:
This seems really inefficient.  How about adding another two keywords that would set the default line/color to black/viridis?

Reply:
I don't understand your suggestion.

Reply:
```
def routine(...., default_cmap = 'viridis', default_cline = 'k'):
```

and then use the defaults in the dict.

Reply:
That way you can change all the colors without having to remember all of the different variable names.

Reply:
How about I check if cmap=str or a dict. if a str then all plots change and if a dict then you pick what changes.

Reply:
because adding more options for the user to have to remember is also annoying and inefficient :p

Reply:
Whichever way you think is best.  From what I see, both methods have downsides.

Reply:
Done